src/Symfony/Component/Scheduler/Scheduler.php

```diff
@@ -81,10 +82,14 @@ public function run(array $options = []): void
                         continue;
                     }
 
-                    $this->handlers[$message::class]($message);
-                    $ran = true;
+                    try {
+                        $this->handlers[$message::class]($message);
+                        $ran = true;
 
-                    $this->dispatcher->dispatch(new PostRunEvent($generator->getSchedule(), $context, $message));
+                        $this->dispatcher->dispatch(new PostRunEvent($generator->getSchedule(), $context, $message));
+                    } catch (\Throwable $error) {
+                        $this->dispatcher->dispatch(new FailureEvent($generator->getSchedule(), $context, $message, $error));
```

Comment:
We need to rethrow the exception, right?
And maybe the FailureEvent can have a way to ignore the error?

Reply:
yes indeed. Maybe I could add a `shouldIgnore` like the one we already have for the `WorkerMessageReceivedEvent`